enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,349 @@
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceType string
+
+const (
+	// The DeviceTypes that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceType = "Disk"
+	// Part represents a device-type of partion
+	Partition DeviceType = "Part"
+)
+
+type DeviceTypeList struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty the default value will be `[disk]`.
+	Devices []DeviceType `json:"devices"`
+}
+
+type DeviceInclusionSpec struct {
+	// DeviceTypeList holds the list of devices that should be used for automatic detection.
+	// If Nil the default detection will be of disks.
+	DeviceTypeList *DeviceTypeList `json:"deviceTypeList"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupPhase string
+
+const (
+	DiscoveringPhase LocalVolumeGroupPhase = "Discovering"
+	FailedPhase      LocalVolumeGroupPhase = "Failed"
+	DiscoveredPhase  Phase                 = "Discovered"
+	ProvisionedPhase LocalVolumeGroupPhase = "Provisioned"
+)
+
+type LocalVolumeGroupStatus struct {
+	// Phase describes the state of the LocalVolumeGroup
+	Phase LocalVolumeGroupPhase `json:"phase,omitempty"`
+
+	// A human-readable message indicating details about why the LocalVolumeGroup is in this state.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Reason is a brief CamelCase string that describes any failure and is meant
+	// for machine parsing and tidy display in the CLI.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// TotalDiscoveredDeviceCount is the count of the total devices which matched the inclusion filter
+	TotalDiscoveredDeviceCount *int32 `json:"totalDiscoveredDeviceCount,omitempty"`
```

Comment:
How will this field get populated? Can each node modify the `LocalVolumeGroup` object and update this? How are you going to keep this updated? You will have to keep track of devices which were "already" used for local-storage vs new devices. IMO this field should be dropped.

Reply:
yeah it would be difficult to expose this. Initially, the plan was to do it via configmap.

Reply:
dropped.

Reply:
should we also remove `TotalProvisionedDeviceCount` ?